storage/encrypting_repository.go

```diff
@@ -14,14 +14,23 @@ import (
 	"github.com/Peripli/service-manager/pkg/types"
 )
 
-// KeyStore interface for encryption key operations
-type KeyStore interface {
+// LockerCreatorFunc is a function building a storage.Locker with a specific advisory index
+type LockerCreatorFunc func(advisoryIndex int) Locker
+
+// Locker provides basic Lock/Unlock functionality
+type Locker interface {
 	// Lock locks the storage so that only one process can manipulate the encryption key. Returns an error if the process has already acquired the lock
 	Lock(ctx context.Context) error
 
+	// TryLock tries to lock the storage so that only one process can manipulate the encryption key. Returns an error if the process has already acquired the lock
+	TryLock(ctx context.Context) error
```

Comment:
huh? this has the same declaration and godoc, from an interface perspective I don't understand which to use

Reply:
Will be revised